### PR TITLE
issue #2375 update statement when onlyUpdate is false;

### LINKED ISSUE #2375: Invalid update SQL generated for HSQL database with different onlyUpdate values in InsertOrUpdateGeneratorHsql.java
## Environment
Liquibase: 4.4.3+, verified on 4.5.0
Database Vendor: HSQL

## Description
Introduced in PR #1563 
Invalid SQL generated for HSQL database in InsertOrUpdateGeneratorHsql.java
when onlyUpdate is "false"
Note: Found when trying to use:

```
  - changeSet:
      id: load-hex-field-data
      author: cyprian
      changes:
        - loadUpdateData:
            encoding: "UTF-8"
            file: "db/data/hex_fields.csv"
            onlyUpdate: "false"
            primaryKey: "id"
            quotchar: "'"
            separator: ","
            tableName: "hex_field"
```

* no need for table name in update part
* no where needed clause in update script

## Steps To Reproduce
changeset file:
```
  - changeSet:
      id: 1
      author: example
      changes:
        - createTable:
            tableName: "hex_field"
            columns:
              - column:
                  name: id
                  type: int
                  constraints:
                    primaryKey: true
                    nullable: false
              - column:
                  name: location_x
                  type: int
              - column:
                  name: location_y
                  type: int
              - column:
                  name: type
                  type: varchar(20)
  - changeSet:
      id: load-hex-field-data
      author: cyprian
      changes:
        - loadUpdateData:
            encoding: "UTF-8"
            file: "db/data/hex_fields.csv"
            onlyUpdate: "false"
            primaryKey: "id"
            quotchar: "'"
            separator: ","
            tableName: "hex_field"
```
CSV file (db/data/hex_fields.csv)
```
id,location_x,location_y,type
1, 1000, 1000,EMPTY
```

## Actual Behavior && Desired Behavior
when onlyUpdate: "false"
```
MERGE INTO hex_field USING (VALUES (1)) ON id = 1 WHEN NOT MATCHED THEN INSERT (id,location_x,location_y,type) VALUES (1, 1000, 1000,' ''EMPTY') WHEN MATCHED THEN UPDATE hex_field SET id = 1,location_x =  1000,location_y =  1000,type = ''EMPTY' WHERE id = 1]
```
should be
```
MERGE INTO hex_field USING (VALUES (1)) ON id = 1 WHEN NOT MATCHED THEN INSERT (id,location_x,location_y,type) VALUES (1, 1000, 1000,' ''EMPTY') WHEN MATCHED THEN UPDATE SET id = 1,location_x =  1000,location_y =  1000,type = 'EMPTY' ]
```
When onlyUpdate: "true" it works as expected and should be
```
UPDATE hex_field SET id = 1,location_x =  1000, location_y =  1000, type = 'EMPTY' WHERE id = '1'
```

### PR DESCRIPTION
**Pull request by devorgpl**
issue #2375 update statement when onlyUpdate is false;
reverted changes from PR #1563 so now it covers both onlyUpdate options

## Environment

**Liquibase Version**: 4.5.0

**Liquibase Integration & Version**: -

**Liquibase Extension(s) & Version**: -

**Database Vendor & Version**: HSQLDB 2.6.0

**Operating System Type & Version**: ALL

## Pull Request Type

- [X] Bug fix (non-breaking change which fixes an issue.)
- [ ] Enhancement/New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
closes #2375 ; Update part of sql generated for updateOnly: true/false options should be different. 

A clear and concise description of the issue being addressed.  Additional guidance [here](https://liquibase.jira.com/wiki/spaces/LB/pages/1274904896/How+to+Contribute+Code+to+Liquibase+Core).
- Describe the actual problematic behavior.
- Ensure private information is redacted.

## Steps To Reproduce

List the steps to reproduce the behavior.
- Please be precise and ensure private information is redacted
- Include things like
  - Files used - sql scripts, changelog file(s), property file(s), config files, POM Files
  - Exact commands used - CLI, maven, gradle, spring boot, servlet, etc.

## Actual Behavior
A clear and concise description of what happens in the software **before** this pull request.
- Include console output if relevant
- Include log files if available.

## Expected/Desired Behavior
A clear and concise description of what happens in the software **after** this pull request.

## Fast Track PR Acceptance Checklist:

- [X] Build is successful and all new and existing tests pass
- [X] Added [Unit Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests)
- [ ] Added [Integration Test(s)](https://liquibase.jira.com/wiki/spaces/LB/pages/1276608569/How+to+Write+Liquibase+Core+Integration+Tests)
- [ ] Added [Test Harness Test(s)](https://github.com/liquibase/liquibase-test-harness/pulls)
- [ ] Documentation Updated

## Need Help?
Come chat with us on our [discord channel](https://discord.com/channels/700506481111597066/700506481572839505)

### CODE REVIEW

File: liquibase-core/src/main/java/liquibase/sqlgenerator/core/InsertOrUpdateGeneratorHsql.java [hunk 1]
```diff
@@ -54,11 +52,13 @@ protected String getElse(Database database) {
 
 	@Override
 	protected String getUpdateStatement(InsertOrUpdateStatement insertOrUpdateStatement, Database database,
-			String whereClause, SqlGeneratorChain sqlGeneratorChain) {
+										String whereClause, SqlGeneratorChain sqlGeneratorChain) throws LiquibaseException {
+
+		if (insertOrUpdateStatement.getOnlyUpdate()) {
```
Comment (devorgpl) [inline liquibase-core/src/main/java/liquibase/sqlgenerator/core/InsertOrUpdateGeneratorHsql.java:57]: Only this code added, everything else is a revert of #1563

**Review by nvoxland — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/sqlgenerator/core/InsertOrUpdateGeneratorHsql.java (hunk 1 above)
Comment (nvoxland) [reply liquibase-core/src/main/java/liquibase/sqlgenerator/core/InsertOrUpdateGeneratorHsql.java:57]: Makes sense, thanks

**Comment by nvoxland** [comment]
The changes look good, thanks. I checked both the the onlyUpdate: true and onlyUpdate: false settings

**Review by suryaaki2 — APPROVED** [review]

**Comment by XDelphiGrl** [comment]
Liquibase respects `onlyUpdate=false` in the `loadUpdateData` changetype for HSQL.

**Special Thanks!** @devorgpl, thanks for the integration test. I validated it passed on the fix branch build.
```
-------------------------------------------------------------------------------
Test set: liquibase.sqlgenerator.core.InsertOrUpdateGeneratorHsqlTest
-------------------------------------------------------------------------------
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.022 s - in liquibase.sqlgenerator.core.InsertOrUpdateGeneratorHsqlTest
```

---

_Verify update-sql generates correct insert for `onlyUpdate=true` in the `loadUpdateData` changetype._ **PASS**

```
-- Changeset pr2377-changelog.yaml::load-hex-field-data::cyprian
-- WARNING The following SQL may change each run and therefore is possibly incorrect and/or invalid:
UPDATE PUBLIC.hex_field SET location_x = ' 1000', location_y = ' 1000', type = 'EMPTY' WHERE id = '1';
```
_Verify update-sql generates correct merge into statement for `onlyUpdate=false` in the `loadUpdateData` changetype._ **PASS**

```
-- Changeset pr2377-changelog.yaml::load-hex-field-data::cyprian
-- WARNING The following SQL may change each run and therefore is possibly incorrect and/or invalid:
MERGE INTO hex_field USING (VALUES (1)) ON id = '1' WHEN NOT MATCHED THEN INSERT (id,location_x,location_y,type) VALUES ('1',' 1000',' 1000','EMPTY') WHEN MATCHED THEN UPDATE SET id = '1',location_x = ' 1000',location_y = ' 1000',type = 'EMPTY';
```
_Verify update with `onlyUpdate=false` in `loadUpdateData` is successful._  **PASS**

```
[2022-02-28 15:53:02] INFO [liquibase.changelog] Table hex_field created
[2022-02-28 15:53:02] INFO [liquibase.changelog] ChangeSet pr2377-changelog.yaml::1::example ran successfully in 11ms
[2022-02-28 15:53:02] INFO [liquibase.changelog] Data loaded from 'data/pr2377.csv' into table 'hex_field'
```

---
**Test Environment**
Liquibase Core: //2375-update-stmt-when-onlyUpdate-false/1220/acd459, Pro: master/405/dee743/
HSQL 2.6.1
[Passing Functional Tests](https://github.com/liquibase/liquibase-pro-tests/actions/runs/1703087557)
[Passing Functional Tests - Oracle](https://github.com/liquibase/liquibase-pro-tests/runs/5366346593?check_suite_focus=true)
Note: An Oracle regression test failed because the test branch needed to be updated; re-ran test to get full green run for this PR.

**Pull request merged**